### PR TITLE
Put debug symbols in releases

### PR DESCRIPTION
**Pull request by expipiplus1**
Closes https://github.com/shader-slang/slang/issues/5267

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -110,12 +110,12 @@ jobs:
...
-              [ -f "build.em/Release/lib/libslang.a" ]
-              [ -f "build.em/Release/lib/libcompiler-core.a" ]
-              [ -f "build.em/Release/lib/libcore.a" ]
+              cmake --build --preset emscripten --config "$cmake_config" --target slang
+              [ -f "build.em/$cmake_config/lib/libslang.a" ]
+              [ -f "build.em/$cmake_config/lib/libcompiler-core.a" ]
+              [ -f "build.em/$cmake_config/lib/libcore.a" ]
           else
-            if [[ "${{ matrix.os }}" =~ "windows" && "${{ matrix.config }}" != "release" ]]; then
+            if [[ "${{ matrix.os }}" =~ "windows" && "${{ matrix.config }}" != "release" && "${{ matrix.config }}" != "releaseWithDebugInfo" ]]; then
```
Comment (jkwak-work) [inline .github/workflows/ci.yml:118]: I think we can keep both cases "release" and "releaseWithDrbugInfo", here.
```
&& "${{ matrix.config }}" != "release" && "${{ matrix.config }}" != "releaseWithDebugInfo"
```

**Review by jkwak-work — DISMISSED** [review]
Looks good to me.

**Comment by csyonghe** [comment]
Have you verified this works on windows? From the build log it is still outputting to the Release folder.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (expipiplus1) [reply .github/workflows/ci.yml:118]: sorry, what do you mean?

Comment (jkwak-work) [reply .github/workflows/ci.yml:118]: Oops. Sorry I read the diff incorrectly.
Please ignore my comment above.

**Comment by expipiplus1** [comment]
> Have you verified this works on windows? From the build log it is still outputting to the Release folder.

Where are you observing this, in CI it's compiling into RelWithDebInfo 
image


I changed things to make RelWithDebInfo the default config btw

**Comment by csyonghe** [comment]
There are a  lot of warnings on windows builds:

```
cl : Command line warning D9025 : overriding '/MD' with '/MT'
```

**Comment by expipiplus1** [comment]
yeah, I'm looking into that now

**Comment by expipiplus1** [comment]
ok, looks like it's fixed

File: clean.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+../slang-ellie/clean.sh
```
Comment (csyonghe) [inline clean.sh:1]: what's this?

**Comment by csyonghe** [comment]
Question: when this lands, what is our prebuilt binary package directory structure? Are things under `bin/RelWithDebInfo/` instead of `bin/Release`. I really don't like `RelWithDebInfo`.

**Comment by expipiplus1** [comment]
image
This has been the layout for a while now, no `Release` or `RelWithDebInfo` either way

In the build tree it'll look like `build/RelWithDebInfo/bin/*` yes. It's very possible with CMake to change that to `build/bin` for example (or `build/Foo/bin`) if that's something you want?

**Review by csyonghe — APPROVED** [review]
On Linux, are the debug symbols embedded into the so? Can we separate the debug symbols into their own files like on windows?

**Comment by csyonghe** [comment]
The package layout is good, no need for changes then.

**Comment by expipiplus1** [comment]
It's definitely possible to do separate debug info on linux (but no on MacOS I think?), but it's not very common in my experience. 

In fact, I think at the moment we're embedding the debug info on Windows as well (sccache doesn't support separate debug info iirc)

**Comment by csyonghe** [comment]
IIRC the binaries with debug info embedded is very big, something about 50MB? Not sure if it actually is a good idea to include it for all of our users, but it is probably still the better choice? We could provide a separate withDebInfo package for Linux and windows x64 only.

**Pull request merged**